components/group/groupSlice.ts

```diff
@@ -108,9 +147,68 @@ export const groupSlice = createSlice({
             console.log(`Create user rejected: ${JSON.stringify(action.error)}`)
             state.loading = false
         })
+        // createNewGroup
+        builder.addCase(fetchRegisteredGroups.pending, (state) => {
+            state.loading = true
+        })
+        builder.addCase(fetchRegisteredGroups.fulfilled, (state, action) => {
+            state.loading = false
+            state.groups = action.payload
+        })
+        builder.addCase(fetchRegisteredGroups.rejected, (state, action) => {
+            console.log(`Create user rejected: ${JSON.stringify(action.error)}`)
+            state.loading = false
+        })
     },
 })
 
 export const groupListSelector = (state: RootState) => state.group.groups
 
 export default groupSlice.reducer
+
+const registeredGroupsSelector = (state: RootState) => state.group.registeredGroups
+
+export const useGroupStorage: () => [RegisteredGroup[], (groups: RegisteredGroup[]) => void] = () => {
+    const registeredGroups = useAppSelector(registeredGroupsSelector)
+    const dispatch = useAppDispatch()
+    const setRegisteredGroups = (registeredGroups: RegisteredGroup[]) => {
+        dispatch(groupSlice.actions.registerNewGroups(registeredGroups))
+    }
+    return [registeredGroups, setRegisteredGroups]
+}
+
+export const useCurrentGroup = () => {
+    const dispatch = useAppDispatch()
+    const allGroups = useAppSelector(groupListSelector)
+    const [registeredGroups, setRegisteredGroups] = useGroupStorage()
+    const router = useRouter()
+    const groupIdFromRouter = router.query.groupId as string
+
+    // if a new group is visited, save it to local storage
+    if (registeredGroups.find(g => g.groupId === groupIdFromRouter) === undefined) {
+        console.log(`Registering a new group: ${groupIdFromRouter}`)
+        const newRegisteredGroup = {
+            groupId: groupIdFromRouter,
+            asUserId: null
+        }
+        setRegisteredGroups([...registeredGroups, newRegisteredGroup])
+        dispatch(fetchRegisteredGroups())
+    }
```

Comment:
Seems like this should be in a `useEffect`. Right now it is called everytime `useCurrentGroup` is called, but it seems like we only care about running this when `registeredGroups` or `groupIdFromRouter` change.